homeassistant/components/ai_task/__init__.py

```diff
@@ -164,3 +206,62 @@ def async_set_preferences(
     def as_dict(self) -> dict[str, str | None]:
         """Get the current preferences."""
         return {key: getattr(self, key) for key in self.KEYS}
+
+
+class ImageView(HomeAssistantView):
+    """View to generated images."""
+
+    url = f"/api/{DOMAIN}/images/{{filename}}"
+    name = f"api:{DOMAIN}/images"
+    requires_auth = False
+
+    async def get(
+        self,
+        request: web.Request,
+        filename: str,
+    ) -> web.Response:
+        """Serve image."""
+        hass = request.app[KEY_HASS]
+        IMAGE_STORAGE = hass.data.setdefault(DATA_IMAGES, {})
+        image_data = IMAGE_STORAGE.get(filename)
+
+        if image_data is None:
+            raise web.HTTPNotFound
+
+        return web.Response(
+            body=image_data.data,
+            content_type=image_data.mime_type,
+        )
+
+
+class ThumbnailView(HomeAssistantView):
```

Comment:
I don't think that we should start with a thumbnail from the start. I wonder if we can leverage the infrastructure from the image upload integration, which automatically generates thumbnails.